### DISCUSSION TITLE
How to treat (undetected) syntax errors in translation strings?

### DISCUSSION THREAD

**Discussion by buhtz** [python-babel/babel#1111] [General]
Hello,

I am an upstream maintainer managing over 40 languages in a Python application. There is one string in my Spanish translation containing to many } and causing exceptions when running. My translation platform (Weblate) was not able to detect that.

How would you suggest to deal with possible errors like that. I can not manually review all translations and also don't have enough skilled reviewers in each of that language.

I think about to somehow test that in a script using every translatable string in every available language.

What do you think?

Best,
Christian

X-Post: https://github.com/izimobil/polib/issues/156

**Comment by akx** [comment]
Could you show an example of the buggy message (e.g. a link to a commit fixing the .po file)?

**Comment by buhtz** [comment]
The commit is to big but
1. `"foo {bar}}"`
2. `"foo {{bar}}"` (this won't throw an Exception but is a typo in most cases)
3. translated: `"{foo} {bar}"` original source: `"foo {bar}"`

I have seen all this cases in my real translations. I have a pending PR fixing the translations and improving my helper script checking such cases in po files.

https://github.com/bit-team/backintime/pull/1829

But there might be a more elegant solution?
